firestore/src/androidTest/java/com/firebase/ui/firestore/FirestoreDataSourceTest.java

```diff
@@ -1,213 +0,0 @@
-package com.firebase.ui.firestore;
-
-import com.firebase.ui.firestore.paging.FirestoreDataSource;
-import com.firebase.ui.firestore.paging.LoadingState;
-import com.firebase.ui.firestore.paging.PageKey;
-import com.google.android.gms.tasks.Tasks;
-import com.google.firebase.firestore.DocumentSnapshot;
-import com.google.firebase.firestore.Query;
-import com.google.firebase.firestore.QuerySnapshot;
-import com.google.firebase.firestore.Source;
-
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.Mock;
-import org.mockito.MockitoAnnotations;
-
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-import java.util.concurrent.CountDownLatch;
-
-import androidx.annotation.Nullable;
-import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
-import androidx.lifecycle.Observer;
-import androidx.paging.PageKeyedDataSource;
-import androidx.test.ext.junit.runners.AndroidJUnit4;
-
-import static org.junit.Assert.assertEquals;
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyLong;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
-
-@RunWith(AndroidJUnit4.class)
-public class FirestoreDataSourceTest {
```

Comment:
Should we still have a version of this test that looks out for the native Paging events? Or do you think these tests have no value anymore?

Reply:
With this upgrade, paging events are no longer provided by the DataSource, instead they're provided by the `PagingDataAdapter`.  I'm trying to figure out how to write tests for the new adapter - once I find out I'll send a PR with the new version of this test.